.github/workflows/codeql.yml

```diff
@@ -0,0 +1,49 @@
+name: "CodeQL"
+
+permissions:
+  contents: read
+
+on:
+  push:
+    branches:
+    - "main"
+  pull_request:
+    branches:
+    - "main"
+  schedule:
+  - cron: '16 11 * * 5'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: 'ubuntu-22.04'
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+        - 'go'
+        - 'javascript-typescript'
+        - 'python'
```

Comment:
unreltedly - ive been thinking we should add some other langs in the envoy setup